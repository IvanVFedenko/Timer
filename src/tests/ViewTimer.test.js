import React from 'react';
import ViewTimer from '../components/viewTimer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import { number, object } from 'prop-types';

Enzyme.configure({ adapter: new Adapter() });

describe('ViewTimer component initial', () => {
  const myComponent = shallow(<ViewTimer />);

  it('render initial', () => {
    expect(myComponent.find('div')).toHaveLength(4);
  });
  it('renders properly', () => {
    expect(myComponent).toMatchSnapshot();
  });
});

describe('testing ViewTimer component ', () => {
  it('snapshot header ', () => {
    const wrapper = shallow(<ViewTimer />);
    expect(wrapper.find('.Countdown-header').text()).toEqual('Countdown');
  });
});

describe('testing functionality', () => {
  it('startTimer callback calling', () => {
    let startTimer = jest.fn();
    const wrapper = shallow(<ViewTimer startTimer={startTimer} />);
    const startButton = wrapper.find('.Button-start');
    startButton.at(0).simulate('click');
    expect(startTimer).toBeCalled();
  });

  it('startFaster callback calling', () => {
    let startFaster = jest.fn();
    const wrapper = shallow(<ViewTimer startFaster={startFaster} />);
    const startButton = wrapper.find('.Button-start');
    startButton.at(1).simulate('click');
    expect(startFaster).toBeCalled();
  });

  it('startFastest callback calling', () => {
    let startFastest = jest.fn();
    const wrapper = shallow(<ViewTimer startFastest={startFastest} />);
    const startButton = wrapper.find('.Button-start');
    startButton.at(2).simulate('click');
    expect(startFastest).toBeCalled();
  });
});
