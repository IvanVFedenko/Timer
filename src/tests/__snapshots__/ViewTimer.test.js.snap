// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTimer component initial renders properly 1`] = `ShallowWrapper {}`;
